src/L2SharedBridge.sol

```diff
@@ -1,183 +1,124 @@
+}
 
-/// @author Matter Labs
+/// @author Sophon
+/// @notice Forked from ML L1SharedBridge contract
 /// @custom:security-contact security@matterlabs.dev
 /// @notice The "default" bridge implementation for the ERC20 tokens. Note, that it does not
 /// support any custom token logic, i.e. rebase tokens' functionality is not supported.
 contract L2SharedBridge is IL2SharedBridge, Initializable {
-    /// @dev The address of the L1 shared bridge counterpart.
-    address public override l1SharedBridge;
-
-    /// @dev Contract that stores the implementation address for token.
-    /// @dev For more details see https://docs.openzeppelin.com/contracts/3.x/api/proxy#UpgradeableBeacon.
-    UpgradeableBeacon public l2TokenBeacon;
+    using SafeERC20 for IERC20;
 
-    /// @dev Bytecode hash of the proxy for tokens deployed by the bridge.
-    bytes32 internal l2TokenProxyBytecodeHash;
+    uint160 constant SYSTEM_CONTRACTS_OFFSET = 0x8000; // 2^15
 
-    /// @dev A mapping l2 token address => l1 token address
-    mapping(address l2TokenAddress => address l1TokenAddress) public override l1TokenAddress;
-
-    /// @dev The address of the legacy L1 erc20 bridge counterpart.
-    /// This is non-zero only on Era, and should not be renamed for backward compatibility with the SDKs.
-    address public override l1Bridge;
+    /// @dev The address of the L1 shared bridge counterpart.
+    address public override l1SharedBridge;
 
     /// @dev Contract is expected to be used as proxy implementation.
     /// @dev Disable the initialization to prevent Parity hack.
-    uint256 immutable ERA_CHAIN_ID;
+    // uint256 immutable ERA_CHAIN_ID; // TODO: check if i need this!
+
+    /// @dev The address of the USDC token on L1.
+    address public immutable L1_USDC_TOKEN;
 
-    constructor(uint256 _eraChainId) {
-        ERA_CHAIN_ID = _eraChainId;
+    /// @dev The address of the USDC token on L2.
+    address public immutable L2_USDC_TOKEN;
+
+    constructor(address _l1UsdcToken, address _l2UsdcToken) {
+        // ERA_CHAIN_ID = _eraChainId; // TODO: checke if we need this!
+        L1_USDC_TOKEN = _l1UsdcToken;
+        L2_USDC_TOKEN = _l2UsdcToken;
         _disableInitializers();
     }
 
     /// @notice Initializes the bridge contract for later use. Expected to be used in the proxy.
     /// @param _l1SharedBridge The address of the L1 Bridge contract.
-    /// @param _l1Bridge The address of the legacy L1 Bridge contract.
-    /// @param _l2TokenProxyBytecodeHash The bytecode hash of the proxy for tokens deployed by the bridge.
-    /// @param _aliasedOwner The address of the governor contract.
-    function initialize(
-        address _l1SharedBridge,
-        address _l1Bridge,
-        bytes32 _l2TokenProxyBytecodeHash,
-        address _aliasedOwner
-    ) external reinitializer(2) {
+    /// _l1Bridge The address of the legacy L1 Bridge contract.
+    function initialize(address _l1SharedBridge) external reinitializer(2) {
```

Comment:
You don't need reinitalizes(2). But its probably ok to have it (I'm not sure)

Reply:
Will double check! Ty